packages/wasm-bindings/src/lib.rs

```diff
@@ -7,6 +7,82 @@ use wasm_bindgen::prelude::*;
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
+/// Modify the currently selected tool in the document state store
+#[wasm_bindgen]
+pub fn select_tool(tool_id: u32) {
+	todo!()
+}
+
+/// Modify the currently selected Document in the editor state store
+// TODO: should it be possible to open multiple documents at once?
+#[wasm_bindgen]
+pub fn select_document(document_id: u32) {
+	todo!()
+}
+
+/// Load a new .gdd file into the editor
+/// Returns a unique document identifier
+#[wasm_bindgen]
+pub fn load_document(raw_data: &[u8]) -> u32 {
+	todo!()
+}
+
+/// Mouse path drawn on the canvas if the mouse is clicked for a longer period of time
+/// (every mouse position between the mouse down and up event)
+#[wasm_bindgen]
+pub fn canvas_mouse_path(mousepath: &[u64], modkeys: u8, ui_element_id: u32) -> u32 {
+	todo!()
+}
+
+/// Notify the editor that the mouse hovers above a panel
+#[wasm_bindgen]
+pub fn panel_enter_hover(panel_id: u32) {
+	todo!()
+}
+
+/// Notify the editor that the mouse leaves a panel
+#[wasm_bindgen]
+pub fn panel_leave_hover(panel_id: u32) {
+	todo!()
+}
+
+/// Notify the editor that a UI-element is clicked
+// TODO: should this function return UI changes?
+#[wasm_bindgen]
+pub fn click_element(ui_element_id: u32, modkeys: u8) {
+	todo!()
+}
+
+/// Update working colors
+#[wasm_bindgen]
+pub fn update_colors(primary_color: u32, secondary_color: u32) {
```

Comment:
Colors will be defined as:

struct Color {
    r: f32,
    g: f32,
    b: f32,
    a: f32,
}

Reply:
```
struct Color {
    r: f32,
    g: f32,
    b: f32,
    a: f32,
}
```